src/Campaigns/Blocks/CampaignDonorsBlock/render.php

```diff
@@ -0,0 +1,261 @@
+            $sortBy = $this->attributes['sortBy'] ?? 'top-donors';
+            $query = new CampaignDonationQuery($this->campaign);
+
+            if ($sortBy === 'top-donors') {
+                $query->select(
+                    'donorIdMeta.meta_value as id',
+                    'SUM(amountMeta.meta_value) as amount',
+                    'donors.name as name'
+                )
+                    ->joinDonationMeta(DonationMetaKeys::DONOR_ID, 'donorIdMeta')
+                    ->joinDonationMeta(DonationMetaKeys::AMOUNT, 'amountMeta')
+                    ->leftJoin('give_donors', 'donorIdMeta.meta_value', 'donors.id', 'donors')
+                    ->groupBy('donorIdMeta.meta_value')
+                    ->orderBy('amount', 'DESC');
+            } else {
+                $query->select(
+                    'donation.ID as donationID',
+                    'donorIdMeta.meta_value as id',
+                    'companyMeta.meta_value as company',
+                    'donation.post_date as date',
+                    'amountMeta.meta_value as amount',
+                    'donors.name as name'
+                )
+                    ->joinDonationMeta(DonationMetaKeys::DONOR_ID, 'donorIdMeta')
+                    ->joinDonationMeta(DonationMetaKeys::COMPANY, 'companyMeta')
+                    ->joinDonationMeta(DonationMetaKeys::AMOUNT, 'amountMeta')
+                    ->leftJoin('give_donors', 'donorIdMeta.meta_value', 'donors.id', 'donors')
+                    ->orderBy('donation.ID', 'DESC');
+            }
+
+            $query->limit($this->attributes['donorsPerPage'] ?? 5);
+
+            if ( ! $this->attributes['showAnonymous']) {
+                $query->joinDonationMeta(DonationMetaKeys::ANONYMOUS, 'anonymousMeta')
+                    ->where('anonymousMeta.meta_value', '0');
+            }
+
+            return $query;
+        }
+
+        /**
+         * @unreleased
+         */
+        private function formatDonorsData(array $donors): array
+        {
+            return array_map(static function ($entry) {
+                if (isset($entry->date)) {
+                    $entry->date = Carbon::parse($entry->date)->diffForHumans();
+                }
+                $entry->amount = Money::fromDecimal($entry->amount, give_get_currency());
+
+                return $entry;
+            }, $donors);
+        }
+
+        /**
+         * @unreleased
+         */
+        private function renderBlockHtml($donors): void
+        {
+            $sortBy = $this->attributes['sortBy'] ?? 'top-donors';
+            $blockTitle = $sortBy === 'top-donors' ? __('Top Donors', 'give') : __(
+                'Recent Donors',
+                'give'
+            );
+            $donateButtonText = $this->attributes['donateButtonText'] ?? __('Join the list', 'give');
+
+            $blockInlineStyles = sprintf(
+                '--givewp-campaign-donors-block__primary-color: %s; --givewp-campaign-donors-block__secondary-color: %s;',
```

Comment:
Can we use `--givewp-primary-color` and `--givewp-secondary-color` here?

Reply:
Renamed on 4b7c5f1ad3454f5c8b1a97a1c0f9a27b45e74c05